presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -440,8 +445,15 @@ public ActualProperties visitJoin(JoinNode node, List<ActualProperties> inputPro
                             .unordered(true)
                             .build();
                 case FULL:
-                    // We can't say anything about the partitioning scheme because any partition of
-                    // a hash-partitioned join can produce nulls in case of a lack of matches
+                    if (probeProperties.getNodePartitioning().isPresent() && buildProperties.getNodePartitioning().isPresent()) {
+                        ImmutableBiMap<VariableReferenceExpression, VariableReferenceExpression> joinMap = node.getCriteria().stream()
+                                .collect(toImmutableBiMap(JoinNode.EquiJoinClause::getLeft, JoinNode.EquiJoinClause::getRight));
+
+                        return ActualProperties.builder()
+                                .global(probeProperties.translate(column -> rewriteToCoalesce(metadata.getTypeManager(), joinMap, column)))
```

Comment:
nit: What does "column" mean here? -- Is this the output after join?

Reply:
It's the partitioning columns